testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -719,8 +754,29 @@ impl SignCoordinator {
                 }
             };
 
+            // look in the nakamoto staging db -- a block can only get stored there if it has
+            // enough signing weight to clear the threshold
+            if let Ok(Some((stored_block, _sz))) = chain_state
```

Comment:
I feel like the order of operations here is maybe wrong - basically it's currently:

- wait up to 500ms for a StackerDB event
- If we get a timeout, start over and wait another 500ms
- only if we got an event, then we check our chainstate for a Nakamoto block

Perhaps we should check our chainstate first, and then poll for a StackerDB event?

Reply:
Yup, you're right